datahub-web-react/src/app/lineage/utils/layoutTree.ts

```diff
@@ -32,6 +32,20 @@ function getParentRelationship(direction: Direction, parent: VizNode | null, nod
     return directionRelationships?.find((r) => r?.entity?.urn === node?.urn);
 }
 
+function firstAppearanceIndices(arr) {
```

Comment:
might be useful to specify why this method exists: because we want to remove duplicates.

Reply:
added a comment